core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileDeleteAction.java

```diff
@@ -88,14 +96,11 @@ public void run(ActionContext context) throws Exception {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (config.fileRegex != null) {
-      filter = new PathFilter() {
-        private final Pattern pattern = Pattern.compile(config.fileRegex);
-
-        @Override
-        public boolean accept(Path path) {
-          return pattern.matcher(path.getName()).matches();
-        }
-      };
+      try {
+        Pattern filter = Pattern.compile(config.fileRegex);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(String.format("File regex is invalid", e));
```

Comment:
The exception should be the 2nd argument passed to the constructor, not an argument of String.format().

`throw new IllegalArgumentException(String.format("File regex '%s' is invalid: %s", config.fileRegex, e.getMessage()), e);`